engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/AvroUtils.scala

```diff
@@ -53,18 +55,20 @@ class AvroUtils(schemaRegistryProvider: SchemaRegistryProvider[_]) extends Seria
 
   private def getOrUpdateSchemaBySubjectAndVersion(subject: String, version: Int): Schema = {
     schemaBySubjectAndVersionCache.getOrElseUpdate((subject, version),
-      schemaRegistryClient.getBySubjectAndVersion(subject, version))
+      handleClientResponse(schemaRegistryClient.getBySubjectAndVersion(subject, version)))
```

Comment:
I see a big problem now. Those methods in `AvroUtils` are not used now and you are parsing schema each time when you ask for them. We need to clean up it ... I think that you should have some cache on top of `ConfluentSchemaRegistryClient` to not invoke `parseSchema` each time. Then all this code won't be needed in `AvroUtils`... Can you take a look for all not used public methods?

Reply:
Take a look at https://github.com/confluentinc/schema-registry/blob/master/client/src/main/java/io/confluent/kafka/schemaregistry/client/CachedSchemaRegistryClient.java#L60 ... schema registry client changed how it caches schemas, maybe we should use correct methods to use it e.g. `getSchemaMetadata` doesn't use caching...

Reply:
Indeed I take a look at wrong version...

Reply:
Good catch - I've added simple cache for it.

Reply:
I see that you are using `SchemaRegistryClient.getSchema` directly from `SchemaRegistryKafkaAvroProvider` and it won't be cached. WDYT about cleaning up this mess that I've introduced by adding cache to `AvroUtils` and introduce new class `CachedParsedSchemaRegistryClient` that will implement `SchemaRegistryClient` and do all this caching in `AvroUtils`? The main purpose of `AvroUtils` was to be used from SpEL + some minor utilities...

Reply:
Changed - take a look.